src/sql/workbench/browser/modelComponents/radioCardGroup.component.ts

```diff
@@ -0,0 +1,179 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+import 'vs/css!./media/card';
+
+import {
+	Component, Inject, ChangeDetectorRef, forwardRef,
+	ElementRef, OnDestroy, Input, ViewChildren, QueryList
+} from '@angular/core';
+
+import { IComponent, IComponentDescriptor, IModelStore, ComponentEventType } from 'sql/workbench/browser/modelComponents/interfaces';
+import { ComponentBase } from 'sql/workbench/browser/modelComponents/componentBase';
+import * as azdata from 'azdata';
+import { getIconClass } from 'sql/workbench/browser/modelComponents/iconUtils';
+import { StandardKeyboardEvent } from 'vs/base/browser/keyboardEvent';
+import { KeyCode } from 'vs/base/common/keyCodes';
+import * as DOM from 'vs/base/browser/dom';
+
+@Component({
+	templateUrl: decodeURI(require.toUrl('./radioCardGroup.component.html'))
+
+})
+export default class RadioCardGroup extends ComponentBase implements IComponent, OnDestroy {
+	@Input() descriptor: IComponentDescriptor;
+	@Input() modelStore: IModelStore;
+	@ViewChildren('cardDiv') cardElements: QueryList<ElementRef>;
+
+	private selectedCard: azdata.RadioCard;
+	private focusedCard: azdata.RadioCard;
+	private iconClasses: { [key: string]: string } = {};
+
+	constructor(
+		@Inject(forwardRef(() => ChangeDetectorRef)) changeRef: ChangeDetectorRef,
+		@Inject(forwardRef(() => ElementRef)) el: ElementRef) {
+		super(changeRef, el);
+	}
+
+	ngOnInit(): void {
+		this.baseInit();
+	}
+
+	setLayout(layout: any): void {
+		this.layout();
+	}
+
+	ngOnDestroy(): void {
+		Object.keys(this.iconClasses).forEach((key) => {
+			DOM.removeCSSRulesContainingSelector(this.iconClasses[key]);
+		});
+		this.baseDestroy();
+	}
+
+	onKeyDown(event: KeyboardEvent): void {
+		if (!this.enabled || this.cards.length === 0) {
+			return;
+		}
+
+		let e = new StandardKeyboardEvent(event);
+		if (e.keyCode === KeyCode.Enter || e.keyCode === KeyCode.Space) {
+			if (this.focusedCard && !this.selectedCard) {
+				this.selectCard(this.focusedCard);
+			}
+			DOM.EventHelper.stop(e, true);
+		}
+		else if (e.keyCode === KeyCode.LeftArrow || e.keyCode === KeyCode.UpArrow) {
```

Comment:
this implies cards are only ever arranged along one dimension; are they never in an grid (where one might want left and up to have different behavior)?

Reply:
currently if we have too many cards to fit in one line it will wrap vertically. see screenshots below. Pressing down key cause the next card to be selected is a little bit unexpected but acceptable especially we want to keep it consistent when there is only one row.
![Screen Shot 2019-12-03 at 9 35 36 AM](https://user-images.githubusercontent.com/13777222/70074844-6455e680-15b0-11ea-9e3a-cef497907658.png)

Reply:
Does pressing down arrow in a normal radio group move to the next one if they're all on a single line?

Reply:
yes, radio card group behaves exactly the same as plain radio buttons

https://www.w3schools.com/tags/tryit.asp?filename=tryhtml5_input_type_radio

![Screen Shot 2019-12-03 at 9 47 47 AM](https://user-images.githubusercontent.com/13777222/70075654-0fb36b00-15b2-11ea-8839-02b0e4753d24.png)